src/tests/JIT/Directed/tailcall/mutual_recursion.fs

```diff
@@ -0,0 +1,252 @@
+
+    static member FifthMethod() = 
+        let firstArg = 10
+        let secondArg = 20
+        let thirdArg = 30
+
+        let point = new Point2D(10.0, 20.0)
+        let secondPoint = new Point2D(30.0, 40.0)
+
+        let rec fifthMethodFirstCallee(iterationCount, firstArg: Point2D, secondArg: Point2D, thirdArg: Point2D, fourthArg: Point2D, fifthArg: Point2D) = 
+            if iterationCount = 0 then
+                100
+            else if iterationCount % 2 = 0 then
+                fifthMethodSecondCallee(iterationCount - 1, firstArg, secondArg, thirdArg, fourthArg, fifthArg)
+            else
+                fifthMethodFirstCallee(iterationCount - 1, firstArg, secondArg, thirdArg, fourthArg, fifthArg)
+
+        and fifthMethodSecondCallee(iterationCount, firstArg: Point2D, secondArg: Point2D, thirdArg: Point2D, fourthArg: Point2D, fifthArg:Point2D) =
+            if iterationCount = 0 then
+                101
+            else if iterationCount % 2 = 0 then
+                fifthMethodSecondCallee(iterationCount - 1, firstArg, secondArg, thirdArg, fourthArg, fifthArg)
+            else
+                fifthMethodFirstCallee(iterationCount - 1, firstArg, secondArg, thirdArg, fourthArg, fifthArg)
+
+        fifthMethodFirstCallee(1000000, point, secondPoint, point, secondPoint, point)
+
+    static member SixthMethod() =
+        let firstArg = 10
+        let secondArg = 20
+        let thirdArg = 30
+
+        let point = new Point2D(10.0, 20.0)
+        let secondPoint = new Point2D(30.0, 40.0)
+
+        let rec sixthMethodFirstCallee(iterationCount, firstArg: Point2D, secondArg: Point2D, thirdArg: Point2D, fourthArg: Point2D, fifthArg: Point2D) = 
+            if iterationCount = 0 then
+                100
+            else if iterationCount % 2 = 0 then
+                sixthMethodSecondCallee(iterationCount - 1, firstArg, secondArg, thirdArg, fourthArg)
+            else
+                sixthMethodFirstCallee(iterationCount - 1, firstArg, secondArg, thirdArg, fourthArg, fifthArg)
+
+        and sixthMethodSecondCallee(iterationCount, firstArg: Point2D, secondArg: Point2D, thirdArg: Point2D, fourthArg: Point2D) =
+            if iterationCount = 0 then
+                101
+            else if iterationCount % 2 = 0 then
+                sixthMethodSecondCallee(iterationCount - 1, firstArg, secondArg, thirdArg, fourthArg)
+            else
+                sixthMethodFirstCallee(iterationCount - 1, firstArg, secondArg, thirdArg, fourthArg, fourthArg)
+
+        sixthMethodFirstCallee(1000000, point, secondPoint, point, secondPoint, point)
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+type Driver() = 
+
+    // Notes:
+    //
+    // Drive all of the different methods that will be called and account for
+    // timing.
+    member this.Start() =
```

Comment:
```suggestion
    member _.Start() =
```

self identifiers are required to be explicit in F# (somewhat annoying), and in many cases they're never used. The typical pattern F# programmers do is use a discard.